buffer/src/main/java/io/netty/buffer/ByteBufOutputStream.java

```diff
@@ -133,7 +134,8 @@ public void writeShort(int v) throws IOException {
     public void writeUTF(String s) throws IOException {
         DataOutputStream out = utf8out;
         if (out == null) {
-            utf8out = out = new DataOutputStream(this);
+            // Suppress a warning since the stream is closed in the close() method
+            utf8out = out = new DataOutputStream(this); // lgtm[java/output-resource-leak]
```

Comment:
I think we should throw if `closed` is true as otherwise we may create a new instance that is never closed again .